### PR TITLE
Deneb Beacon APIs

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

migrates the following APIs from eip4844 to deneb-integration

  - [x] `getBlockV2` json
  - [x] `getBlockV2` ssz
  - [x] `getBlindedBlock` json
  - [x] `getBlindedBlock` ssz
  - [x] `publishBlock` json
  - [ ] `publishBlock` ssz (will be handled in separate PR)
  - [x] `publishBlindedBlock` json
  - [ ] `publishBlindedBlock` ssz (will be handled in separate PR)
  - [x] `produceBlindedBlock` json
  - [ ] `produceBlindedBlock` ssz (will be handled in separate PR)
  - [x]  `produceBlockV2` json
  - [ ] `produceBlockV2` ssz (will be handled in separate PR)

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/rpc/eth/beacon/blinded_blocks.go [hunk 1]
```diff
@@ -130,6 +139,15 @@ func (bs *Server) GetBlindedBlockSSZ(ctx context.Context, req *ethpbv1.BlockRequ
 	if !errors.Is(err, consensus_types.ErrUnsupportedField) {
 		return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
 	}
+	result, err = bs.getBlindedSSZBlockDeneb(ctx, blk)
+	if result != nil {
+		result.Finalized = bs.FinalizationFetcher.IsFinalized(ctx, blkRoot)
+		return result, nil
+	}
+	// ErrUnsupportedGetter means that we have another block type
```
Comment (saolyn) [inline beacon-chain/rpc/eth/beacon/blinded_blocks.go:147]: Same question as above ^

File: beacon-chain/rpc/eth/beacon/blinded_blocks.go [hunk 2]
```diff
@@ -75,6 +75,15 @@ func (bs *Server) GetBlindedBlock(ctx context.Context, req *ethpbv1.BlockRequest
 	if !errors.Is(err, consensus_types.ErrUnsupportedField) {
 		return nil, status.Errorf(codes.Internal, "Could not get blinded block: %v", err)
 	}
+	result, err = bs.getBlindedBlockDeneb(ctx, blk)
+	if result != nil {
+		result.Finalized = bs.FinalizationFetcher.IsFinalized(ctx, blkRoot)
+		return result, nil
+	}
+	// ErrUnsupportedGetter means that we have another block type
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply beacon-chain/rpc/eth/beacon/blinded_blocks.go:83]: seems like the wrong error seems like it reworded

File: proto/eth/v2/BUILD.bazel [hunk 3]
```diff
@@ -92,7 +96,9 @@ go_library(
 ssz_proto_files(
     name = "ssz_proto_files",
     srcs = [
+        "beacon_block.proto",
```
Comment (james-prysm) [inline proto/eth/v2/BUILD.bazel:99]: want to make sure doing something like this makes sense, that beacon block can be moved here to make use of those variables for setting commitment length and blob length in proto

File: beacon-chain/rpc/apimiddleware/structs.go [hunk 4]
```diff
@@ -586,6 +706,25 @@ type ExecutionPayloadCapellaJson struct {
...
+	ReceiptsRoot  string            `json:"receipts_root" hex:"true"`
+	LogsBloom     string            `json:"logs_bloom" hex:"true"`
+	PrevRandao    string            `json:"prev_randao" hex:"true"`
+	BlockNumber   string            `json:"block_number"`
+	GasLimit      string            `json:"gas_limit"`
+	GasUsed       string            `json:"gas_used"`
+	TimeStamp     string            `json:"timestamp"`
+	ExtraData     string            `json:"extra_data" hex:"true"`
+	BaseFeePerGas string            `json:"base_fee_per_gas" uint256:"true"`
+	ExcessDataGas string            `json:"excess_data_gas" uint256:"true"`
```
Comment (terencechain) [inline beacon-chain/rpc/apimiddleware/structs.go:722]: Don't forget `DataGasUsed` field. Same with header

File: beacon-chain/rpc/apimiddleware/structs.go [hunk 5]
```diff
@@ -403,48 +404,111 @@ type SignedBlindedBeaconBlockContainerJson struct {
...
+	Signature string           `json:"signature" hex:"true"`
+}
+
+type BlobSidecarJson struct {
+	BlockRoot       string `json:"block_root" hex:"true"`
+	Index           string `json:"index"`
+	Slot            string `json:"slot"`
+	BlockParentRoot string `json:"block_parent_root" hex:"true"`
+	ProposerIndex   string `json:"proposer_index"`
+	Blob            string `json:"blob" hex:"true"`                // pattern: "^0x[a-fA-F0-9]{262144}$" ssz-max:"4"
```
Comment (terencechain) [inline beacon-chain/rpc/apimiddleware/structs.go:466]: I don't think `ssz-max:"4"` comment is correct. This is just a single blob right?

File: proto/eth/v2/blobs.proto [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        bytes signature = 2 [(ethereum.eth.ext.ssz_size) = "96"];
+}
+
+message BlobIdentifier {
+        bytes block_root = 1 [(ethereum.eth.ext.ssz_size) = "32"];
+        uint64 index = 2;
+}
+
+message SignedBlindedBlobSidecar {
+  BlindedBlobSidecar message = 1 [(ethereum.eth.ext.ssz_max) = "4"];
```
Comment (terencechain) [inline proto/eth/v2/blobs.proto:53]: The max should be 6 now

**Review by terencechain — APPROVED** [review]

**Pull request merged**